Sources/Common/Background Queue/QueueRunRequest.swift

```diff
@@ -38,27 +38,35 @@ public class CioQueueRunRequest: QueueRunRequest {
     }
 
     private func startNewRequestRun() {
-        let inventory = storage.getInventory()
-
-        runTasks(queueInventory: inventory, queryTotalNumberTasks: inventory.count)
+        runTasks(lastRanTask: nil)
     }
 
     // Disable swiftlint because function at this time isn't too complex to need to make it smaller.
     // Many of the lines of this function are logging related.
     // swiftlint:disable:next function_body_length
     private func runTasks(
-        queueInventory: [QueueTaskMetadata],
-        queryTotalNumberTasks: Int,
+        lastRanTask: QueueTaskMetadata?,
         lastFailedTask: QueueTaskMetadata? = nil
     ) {
-        guard let nextTaskToRunInventoryItem = queryRunner.getNextTask(queueInventory, lastFailedTask: lastFailedTask)
+        // get the inventory before running each task. If a task was added to the queue while the last task was being
+        // executed, we can assert that new task will execute during this run.
+        let queueInventory = storage.getInventory()
+
+        guard let nextTaskToRunInventoryItem = queryRunner.getNextTask(
+            queueInventory,
+            lastRanTask: lastRanTask,
+            lastFailedTask: lastFailedTask
+        )
         else {
             // we hit the end of the current inventory. Done!
-            logger.debug("queue out of tasks to run.")
```

Comment:
This log is helpful. Any reason that you thought of removing it ?

Reply:
Feel free to resolve conversation if I understood it right in the comment [below](https://github.com/customerio/customerio-ios/pull/220/files#r1033448843)!